openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -829,6 +838,46 @@ def retrieve(self, request, lib_key_str, pk=None):
         return Response(ContentLibraryBlockImportTaskSerializer(import_task).data)
 
 
+# Library Collections Views
+# =============
+
+@method_decorator(non_atomic_requests, name="dispatch")
+@view_auth_classes()
+class LibraryCollectionsRootView(GenericAPIView):
+    """
+    Views to list and create library collections.
+    """
+
+    # TODO Implement list collections
+
+    def post(self, request, lib_key_str):
+        """
+        Create a new collection library.
+        """
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+        api.require_permission_for_library_key(library_key, request.user, permissions.CAN_EDIT_THIS_CONTENT_LIBRARY)
+        serializer = LibraryCollectionCreationSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        library = api.get_library(library_key)
+        title = serializer.validated_data['title']
+
+        key = slugify(title)
+
+        try:
+            result = authoring_api.create_collection(
+                learning_package_id=library.learning_package_id,
+                key=key,
+                title=title,
+                description=serializer.validated_data['description'],
+                created_by=request.user.id,
+            )
+        except IntegrityError:
+            return Response(status=status.HTTP_409_CONFLICT)
```

Comment:
As the user can't directly edit the slug, do you think it is worth an approach like this?
```suggestion
        attempt = 0
        result = None
        while not result:
            modified_key = key if attempt = 0 else key + str(attempt)
            try:
                result = authoring_api.create_collection(
                    learning_package_id=library.learning_package_id,
                    key=modified_key,
                    title=title,
                    description=serializer.validated_data['description'],
                    created_by=request.user.id,
                )
            except IntegrityError:
                attempt += 1
```

Reply:
@rpenido Thanks! Updated here: https://github.com/openedx/edx-platform/pull/35447/commits/44144c4b443b5063d31a52a56eb2a1a46e054c78